src/bin/server.cpp

```diff
@@ -27,6 +39,43 @@ int main(int argc, char* argv[]) {
     return 0;
   }
 
+  /**
+    * The following code handles the parameter `benchmark_data` which allows users to generate benchmark data when
+    * starting the hyrise server. This is not an ideal solution, but due to several users requests and our goal to
+    * facilitate easy evaluation of hyrise, we decided to integrate the data generation in to the server nonetheless.
+    */
+  if (parsed_options.count("benchmark_data")) {
+    auto benchmark_data_arg = parsed_options["benchmark_data"].as<std::string>();
+    std::vector<std::string> bechmark_data_config;
+
+    boost::trim_if(benchmark_data_arg, boost::is_any_of(":"));
+    boost::replace_all(benchmark_data_arg, "-", "");
+    boost::to_lower(benchmark_data_arg);
+    boost::split(bechmark_data_config, benchmark_data_arg, boost::is_any_of(":"), boost::token_compress_on);
+    Assert(bechmark_data_config.size() < 3,
+           "Malformed input for benchmark data generation. Expecting a benchmark "
+           "name and a sizing factor.");
+
+    const auto benchmark_name = bechmark_data_config[0];
+    const auto sizing_factor = boost::lexical_cast<float, std::string>(bechmark_data_config[1]);
+
+    Assert(benchmark_name == "tpch" || benchmark_name == "tpcds" || benchmark_name == "tpcc",
+           "Benchmark data generation is only supported for TPC-C, TPC-DS, and TPC-H.");
+
+    auto config = std::make_shared<opossum::BenchmarkConfig>(opossum::BenchmarkConfig::get_default_config());
+    config->cache_binary_tables = true;
+    if (benchmark_name == "tpcc") {
+      config->cache_binary_tables = false;  // Not yet supported for TPC-C
+      opossum::TPCCTableGenerator{static_cast<uint32_t>(sizing_factor), config}.generate_and_store();
+    } else if (benchmark_name == "tpcds") {
+      opossum::TPCDSTableGenerator{static_cast<uint32_t>(sizing_factor), config}.generate_and_store();
+    } else if (benchmark_name == "tpch") {
+      opossum::TPCHTableGenerator{sizing_factor, config}.generate_and_store();
+    } else {
+      Fail("Unexpected benchmark name passed in parameter 'benchmark_data'.");
+    }
+  }
```

Comment:
Move this into a separate function? You wouldn't want someone who just wants to understand the server to read this.

Reply:
I had the parsing stuff in CLIConfigParser, but leaves all the benchmark data generation still in the server. Putting the generation there as well doesn't make sense IMO.
Shall I create a benchmark helper? 

I'd prefer moving into an anonymous namespace, but that would leave it in the server.cpp.

Reply:
Anonymous namespace is more than sufficient.